Packs/DeveloperTools/Scripts/WaitAndCompleteTask/WaitAndCompleteTask.py

```diff
@@ -0,0 +1,183 @@
+from time import sleep
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+from typing import Dict, Any
+import traceback
+
+POSSIBLE_STATES = ['New', 'InProgress', 'Completed', 'Waiting', 'Error', 'Skipped', 'Blocked']
+
+''' STANDALONE FUNCTION '''
+
+
+def get_incident_tasks_by_state(incident_id: int, task_states: Optional[list] = None) -> list:
+    """
+
+    Args:
+        incident_id: Incident id to pull its tasks.
+        task_states: States to get. None to get all tasks with all tasks.
+
+    Returns:
+        Tasks with given states related to given incident.
+    """
+    args: Dict[str, Any] = {
+        'incidentId': incident_id
+    }
+    # leave states empty to get all tasks
+    if task_states:
+        args['states'] = ','.join(task_states)
+    try:
+        raw_response = demisto.executeCommand('DemistoGetIncidentTasksByState', args=args)
+        return raw_response[0].get("Contents")
+    except Exception as e:
+        demisto.debug(f'Failed to execute script: DemistoGetIncidentTasksByState. Error: {e}')
+        return []
+
+
+def complete_task_by_id(task_id, task_parent_id, incident_id, complete_option=None) -> str:
+    """
+
+    Args:
+        task_id: task's id to complete
+        task_parent_id: parent playbook task id to complete
+        incident_id: relevant incident
+        complete_option: path option to choose
+
+    Returns:
+        taskComplete automation result
+    """
+    args = assign_params(
+        id=task_id,
+        parentPlaybookID=task_parent_id,
+        incidentId=incident_id,
+        input=complete_option
+    )
+    try:
+        raw_response = demisto.executeCommand('taskComplete', args=args)
+        return raw_response[0].get("Contents")
+    except Exception as e:
+        demisto.debug(f'Failed to execute script: taskComplete. Error: {e}')
+        return ''
```

Comment:
please handle failure of the command gracefully

Reply:
what do you mean ?

Reply:
Fixed, hope that I understood you right.